### ISSUE TITLE
DPP- generate function needs to mask logits when a certain sequence in batch reaches EOS

### ISSUE THREAD

**Issue by Ssukriti** [foundation-model-stack/aiu-fms-testing-utils#181]
**** Problem description ******

When running DPP, generate function from utils is used https://github.com/foundation-model-stack/aiu-fms-testing-utils/blob/main/aiu_fms_testing_utils/utils/paged.py#L31

the generate function stops when all sequences of a batch reached EOS.  However, if one sequence reaches EOS earlier, we are not masking out the logits for that sequence, generation does not stop, and the scores are being used in downstream calculation , increasing CE loss threshold. This also results in more tokens being generated after EOS is reached for some sequences, polluting CPU validation data

**** Solution wanted ***

- we want a simple solution to avoid logits for finished sequences to be used in downstream loss calculation

- we want to avoid tokens after EOS for finished sequences in the CPU validation data

NOTE: we are really looking for simplest solution as this is a test script to get model predictions and not production code. (few inefficiencies in performance are acceptable)

**** Reference scripts ****

HF generation utils: pads finished sequences with PAD tokens to avoid further generation https://github.com/huggingface/transformers/blob/main/src/transformers/generation/utils.py#L2919
Can also refer to FMS/VLLM if needed
